ruby.c

```diff
@@ -36,6 +36,13 @@
 
 #if defined(HAVE_FCNTL_H)
 # include <fcntl.h>
+# if defined(__wasi__)
+// workaround: redefine O_NONBLOCK to use it in #if condition because the original
+// definition is "((__wasi_fdflags_t)(1 << 2))" and __wasi_fdflags_t is unavailable
+// as #if condition expression.
+#  undef O_NONBLOCK
+#  define O_NONBLOCK ((1 << 2))
+# endif
```

Comment:
I hope this part is no longer needed by 2dff82bfca0226e335a446715b94670b5af85108.

Reply:
It works! Thanks :)